vault/core_util.go

```diff
@@ -55,6 +56,26 @@ func (c *Core) setupReplicationResolverHandler() error {
 	return nil
 }
 
+// GetCoreConfigInternal returns the server configuration
+// in struct format.
+func (c *Core) GetCoreConfigInternal() *server.Config {
+	conf := c.rawConfig.Load()
+	if conf == nil {
+		return nil
+	}
+	return conf.(*server.Config)
+}
+
+// SanitizedConfig returns a sanitized version of the current config.
+// See server.Config.Sanitized for specific values omitted.
+func (c *Core) SanitizedConfig() map[string]interface{} {
+	conf := c.rawConfig.Load()
+	if conf == nil {
+		return nil
+	}
+	return conf.(*server.Config).Sanitized()
```

Comment:
is it possible/likely for `*server.Config` to be nil?

Reply:
This just asserts that conf is of type `*server.Config` -- if conf is nil, the above nil check will catch it.

Reply:
ah okay, is it safe to do the `.Sanitized()` for any realistic case here?

I guess I'm concerned because if we're using this as an assertion on a `*server.Config` and the underlying object isn't `*server.Config`, we will panic

basically what i'm asking is: is this safe? if so, go ahead and mark resolved!